scripts/h5toms.py

```diff
@@ -520,3 +529,140 @@ def _create_uvw(a1, a2, times):
+
+                    # get the solution data to write to the main table
+                    # stack in time for storing in caltable
+                    solutions_to_write = np.rollaxis(solvals[0],-1,0)
+                    for sol_iter in solvals[1:]:
+                        solutions_to_write = np.vstack([solutions_to_write,np.rollaxis(sol_iter,-1,0)])
+                    solutions_to_write = np.array(solutions_to_write)
+
+                    # add extra pseudo channel axis to non-B solutions
+                    if sol != 'B':
+                        solutions_to_write = solutions_to_write[:,np.newaxis,:]
+                    # MS's store delays in nanoseconds
+                    if sol == 'K':
+                        solutions_to_write = 1.0e9*solutions_to_write
+
+                    times_to_write = np.repeat(sol_mjd, nants)
+                    antennas_to_write = np.tile(antlist_indices, ntimes)
+                    scans_to_write = np.repeat(range(len(sol_mjd)), nants)
+                    # write the main table
+                    main_cal_dict = ms_extra.populate_caltable_main_dict(times_to_write, solutions_to_write, antennas_to_write, scans_to_write)
+                    ms_extra.write_rows(caltable, main_cal_dict, verbose=options.verbose)
+
+                    # create and write subtables
+                    subtables = ['OBSERVATION', 'ANTENNA', 'FIELD', 'SPECTRAL_WINDOW', 'HISTORY']
+                    subtable_key = [(os.path.join(caltable.name(), st)) for st in subtables]
+
+                    # Add subtable keywords and create subtables
+                    # ------------------------------------------------------------------------------
+                    # # this gives an error in casapy:
+                    # # *** Error *** MSObservation(const Table &) - table is not a valid MSObservation
+                    # for subtable, subtable_location in zip(subtables, subtable_key)
+                    #     ms_extra.open_table(subtable_location, tabledesc=ms_extra.ms_desc[subtable])
+                    #     caltable.putkeyword(subtable, 'Table: {0}'.format(subtable_location))
+                    # # write the static info for the table
+                    # ms_extra.write_dict(caltable_dict, caltable.name(), verbose=options.verbose)
+                    # ------------------------------------------------------------------------------
+                    # instead try just copying the main table subtables
+                    #   this works to plot the data casapy, but the solutions still can't be applied in casapy...
+                    for subtable, subtable_location in zip(subtables, subtable_key):
+                        main_subtable = ms_extra.open_table(main_table.name()+'/'+subtable)
+                        main_subtable.copy(subtable_location, deep=True)
+                        caltable.putkeyword(subtable, 'Table: {0}'.format(subtable_location))
```

Comment:
Ja they seem to be critical (without them it says in some cryptic way that the MS is broken)